flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/SourceOperator.java

```diff
@@ -438,6 +499,11 @@ public void snapshotState(StateSnapshotContext context) throws Exception {
     @Override
     public CompletableFuture<?> getAvailableFuture() {
         switch (operatingMode) {
+            case WAITING_FOR_ALIGNMENT:
+                return availabilityHelper.update(
```

Comment:
nit: is it possible that we are in the `WAITING_FOR_ALIGNMENT` state and the future is `null`?

Reply:
nope, good catch. I guess I got carried away coping pasting the code.